overlord/devicestate/devicestate_test.go

```diff
@@ -801,6 +803,12 @@ func (s *deviceMgrSuite) TestDoRequestSerialErrorsOnNoHost(c *C) {
 		c.Skip("cannot run test when http proxy is in use, the error pattern is different")
 	}
 
+	// check internet access
+	_, err := net.LookupHost("nowhere.nowhere.test")
+	if err != nil && strings.Index(err.Error(), "no such host") == -1 {
```

Comment:
If we need to merge this, let's please use strings.Contains. But the overall string check here seems bogus. If this error message changes for whatever reason this test will be disabled forever and nobody will notice.

Reply:
There is no other way because Go does not use error types, at least not for the net library.